lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+      #
+      # @param stack_id [String] stack_id from the terraforn-runner job
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def stop_async(stack_id)
+        cancel_stack_job(stack_id)
+      end
+
+      # Runs a template, waits until it terraform-runner job completes, via terraform-runner api
+      #
+      # @param input_vars [Hash] Hash with key/value pairs that will be passed as input variables to the
+      #        terraform-runner run
+      # @param template_path [String] Path to the template we will want to run
+      # @param tags [Hash] Hash with key/values pairs that will be passed as tags to the terraform-runner run
+      # @param credentials [Array] List of Authentication object ids to provide to the terraform run
+      # @param env_vars [Hash] Hash with key/value pairs that will be passed as environment variables to the
+      #        terraform-runner run
+      # @return [Terraform::Runner::Response] Response object with final result of terraform run
+      def run(input_vars, template_path, tags: nil, credentials: [], env_vars: {})
+        _log.debug("Run template: #{template_path}")
+        create_stack_job_and_wait_until_completes(
+          template_path,
+          :input_vars  => input_vars,
+          :tags        => tags,
+          :credentials => credentials,
+          :env_vars    => env_vars
+        )
+      end
+
+      # Fetch terraform-runner job result/status by stack_id
+      #
+      # @param stack_id [String] stack_id from the terraforn-runner job
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def fetch_result_by_stack_id(stack_id)
+        retrieve_stack_job(stack_id)
+      end
+
+      # =================================================
+      # TerraformRunner Stack-API interaction methods
+      # =================================================
+      private
+
+      def server_url
+        ENV['TERRAFORM_RUNNER_URL'] || 'https://opentofu-runner:27000'
+      end
+
+      def server_token
+        ENV.fetch('TERRAFORM_RUNNER_TOKEN', nil)
+      end
+
+      def stack_job_interval_in_secs
+        ENV['TERRAFORM_RUNNER_STACK_JOB_CHECK_INTERVAL'].to_i
+      rescue
+        10
+      end
+
+      def stack_job_max_time_in_secs
+        ENV['TERRAFORM_RUNNER_STACK_JOB_MAX_TIME'].to_i
+      rescue
+        120
+      end
+
+      # Create to paramaters as used by terraform-runner api
+      #
+      # @param vars [Hash] Hash with key/value pairs that will be passed as input variables to the
+      #        terraform-runner run
+      # @return [Array] Array of {:name,:value}
+      def convert_to_cam_parameters(vars)
+        parameters = []
+        vars&.each do |key, value|
+          parameters.push(
+            {
+              :name  => key,
+              :value => value
+            }
+          )
+        end
+        parameters
```

Comment:
This looks like you can just use `.map` here
```suggestion
        vars&.map do |key, value|
          {
            :name  => key,
            :value => value
          }
        end
```

Not sure how much you care about returning `[]` if `vars` is `nil`, you could `return [] if vars.nil?` or `vars.to_h.map`

Reply:
done